staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/admission.go

```diff
@@ -196,14 +227,45 @@ func (a *GenericAdmissionWebhook) Admit(attr admission.Attributes) error {
 	hooks := hookConfig.ExternalAdmissionHooks
 	ctx := context.TODO()
 
-	errCh := make(chan error, len(hooks))
-	wg := sync.WaitGroup{}
-	wg.Add(len(hooks))
+	var relevantHooks []*v1alpha1.ExternalAdmissionHook
 	for i := range hooks {
+		if shouldCallHook(&hooks[i], attr) {
+			relevantHooks = append(relevantHooks, &hooks[i])
+		}
+	}
+
+	if len(relevantHooks) == 0 {
+		// no matching hooks
+		return nil
+	}
+
+	// convert the object to the external version before sending it to the webhook
+	versionedAttr := versioned{
+		Attributes: attr,
+	}
+	if attr.GetOldObject() != nil {
+		out, err := a.convertToGVK(attr.GetOldObject(), attr.GetKind())
```

Comment:
// convert to GVK of apiserver endpoint for now.
// TODO: convert to GVK supported by webhook

Reply:
I think "convert to GVK of apiserver endpoint" is what we settled on in the discussion of https://docs.google.com/document/d/1BT8mZaT42jVxtC6l14YMXpUq0vZc6V5MPf_jnzDMMcg/edit#heading=h.5irk4csrpu0y.

Reply:
Not sure this is what I get from the document:

> After some discussions here and in the sig-api-machinery meeting, we decided to drop the conversion requirement and let the user specify the exact subject they are interested in. Thus the negotiation protocol we originally discussed here moved to Alternatives Considered section. The only approved section of this design doc is AdmissionResponse type described in the next sub-section.

This sounds like the GVK will be specified in the webhook registration object.